### PR TITLE
LibGUI: Shift+Tab unindents line

### PR DESCRIPTION
**Pull request by theeyeofcthulhu**
Unindent the current line if Shift+Tab is pressed and no selection is made.

### CODE REVIEW

**Comment by LucasChollet** [comment]
It is required to use a capitalized letter after the colon on the commit message.
Please also squash both commits, the first one will otherwise add dead code.

File: Userland/Libraries/LibGUI/TextEditor.cpp [hunk 1]
```diff
@@ -1050,6 +1055,20 @@ void TextEditor::unindent_selection()
...
+{
+    if (current_line().first_non_whitespace_column() != 0) {
+        auto const space = current_line().leading_spaces();
+        auto const width = space < m_soft_tab_width ? space : m_soft_tab_width;
+
+        auto const temp_column = m_cursor.column();
+
+        execute<UnindentSelection>(width, TextRange({ m_cursor.line(), 0 }, { m_cursor.line(), line(m_cursor.line()).length() }));
+
+        set_cursor({ m_cursor.line(), temp_column <= width ? 0 : temp_column - width });
```
Comment (martinfalisse) [inline Userland/Libraries/LibGUI/TextEditor.cpp:1068]: It seems like you should be able to just do temp_column - width

File: Userland/Libraries/LibGUI/TextEditor.cpp [hunk 2]
```diff
@@ -1050,6 +1055,20 @@ void TextEditor::unindent_selection()
     }
 }
 
+void TextEditor::unindent_line()
+{
+    if (current_line().first_non_whitespace_column() != 0) {
+        auto const space = current_line().leading_spaces();
+        auto const width = space < m_soft_tab_width ? space : m_soft_tab_width;
```
Comment (martinfalisse) [inline Userland/Libraries/LibGUI/TextEditor.cpp:1062]: I would search for a more descriptive variable name .. "unindent_width" ?

File: Userland/Libraries/LibGUI/TextEditor.cpp [hunk 3]
```diff
@@ -1050,6 +1055,20 @@ void TextEditor::unindent_selection()
     }
 }
 
+void TextEditor::unindent_line()
+{
+    if (current_line().first_non_whitespace_column() != 0) {
+        auto const space = current_line().leading_spaces();
```
Comment (martinfalisse) [inline Userland/Libraries/LibGUI/TextEditor.cpp:1061]: This abstraction seems unnecessary to me. IMO it makes the code harder to read

**Review by martinfalisse — COMMENTED** [review]
Looking at previous commits for these files you should be able to get away with simply "LibGUI: Shift+Tab ..", which would fix the commit lint error. :)

**Pull request merged**